pkg/util/cmd/exit.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"context"
+	"errors"
+	"runtime"
+)
+
+// SoftExit sets the exit code to 1 if the error is not a context.Canceled
+// error. It then calls runtime.Goexit() which means that all defer calls
+// will finish before exiting the command. This allows us to cleanup any
+// residual resources when exiting (even on error).
+func SoftExit(err error) {
```

Comment:
`SoftExit` seems like the wrong name to be using here... Can we use `GracefullyExit` instead?

Reply:
renamed it